src/v/cluster/archival/ntp_archiver_service.cc

```diff
@@ -802,6 +826,206 @@ ss::future<> ntp_archiver::upload_until_term_change() {
+          sync_timeout);
+        if (!is_synced.has_value()) {
+            // This can happen on leadership changes, or on timeouts waiting
+            // for stm to catch up: in either case, we should re-check our
+            // loop condition: we will drop out if lost leadership, otherwise
+            // we will end up back here to try the sync again.
+            continue;
+        }
+
+        auto [target_size, min_size] = get_segment_size_limits();
+        upload_candidate_search_parameters search_params(
+          _ntp,
+          _start_term,
+          target_size,
+          min_size,
+          // TODO: use proper quota
+          std::nullopt,
+          std::nullopt,
+          false,
+          manifest_upload_required());
+
+        auto upload_candidate_list = co_await _ops->find_upload_candidates(
+          _rtcnode, search_params);
+
+        auto throttle_upload_loop = [this, &carried_over_backoff]() {
+            carried_over_backoff = carried_over_backoff.value_or(0ms)
+                                   + carried_over_backoff.value_or(
+                                     _initial_backoff());
+            carried_over_backoff = std::clamp(
+              carried_over_backoff.value(), _initial_backoff(), _max_backoff());
+        };
+
+        if (upload_candidate_list.has_error()) {
+            if (
+              upload_candidate_list.error() == error_outcome::not_enough_data) {
+                vlog(_rtclog.debug, "Not enough data to start upload");
+            } else {
+                vlog(
+                  _rtclog.error,
+                  "Upload candidate can't be created: {}",
+                  upload_candidate_list.error());
+            }
+            throttle_upload_loop();
+            continue;
+        }
+
+        auto upload_list = co_await _ops->schedule_uploads(
+          _rtcnode,
+          std::move(upload_candidate_list.value()),
+          manifest_upload_required());
+        if (upload_list.has_error()) {
+            vlog(
+              _rtclog.error,
+              "Failed to schedule uploads: {}",
+              upload_list.error());
+            throttle_upload_loop();
+            continue;
+        }
+
+        // If the manifest was uploaded in parallel with segments we want to
+        // update projected clean offset and last upload timestamp.
+        if (upload_list.value().manifest_clean_offset != model::offset{}) {
+            _projected_manifest_clean_at
+              = upload_list.value().manifest_clean_offset;
+            _last_manifest_upload_time = ss::lowres_clock::now();
+        }
+
+        auto admit_result = co_await _ops->admit_uploads(
+          _rtcnode, std::move(upload_list.value()));
+        if (admit_result.has_error()) {
+            vlog(
+              _rtclog.error,
+              "Failed to admit uploads: {}",
+              admit_result.error());
+            throttle_upload_loop();
```

Comment:
Should this be demoted to info or even debug in the case leadership has moved? Seems like it'd be prone to test flakiness (presuming leadership change is an error code)